torchao/utils.py

```diff
@@ -281,6 +282,30 @@ def unwrap_tensor_subclass(model, filter_fn=None):
         unwrap_tensor_subclass(child)
     return model
 
+class TorchAOBaseTensor(torch.Tensor):
+    """A util tensor subclass that provides commonly used functions
+    """
+    def _get_to_kwargs(self, *args, **kwargs):
+        # `torch._C._nn._parse_to` can't handle `layout` argument
+        for arg in args:
+            if isinstance(arg, torch.layout):
+                args.remove(arg)
+        if "layout" in kwargs:
+            kwargs.pop("layout")
+        device, dtype, _, memory_format = torch._C._nn._parse_to(*args, **kwargs)
+        device = self.device if device is None else device
+        dtype = self.dtype if dtype is None else dtype
+        memory_format = (
+            memory_format if memory_format is not None else torch.preserve_format
+        )
```

Comment:
I have wondered about this for some time. Does it make sense to have `memory_format` applied to inner tensor? From what I understand, it's mainly for channels_last in convolution (https://pytorch.org/docs/stable/tensor_attributes.html#torch-memory-format). Maybe we can skip `memory_format` argument also?

Reply:
if the current tests pass, I think should be good for now!